packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/__tests__/barrier-selector.spec.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import BarrierSelector from '../barrier-selector';
+import TradeParams from '../../../../../Containers/trade-params';
+
+const mock_props = {
+    barrier_1: '16',
+    onChange: () => {
+        mock_props.barrier_1 = '33';
+    },
+    setHoveredBarrier: jest.fn(),
+    turbos_barrier_choices: ['16', '33', '40'],
+};
+
+jest.mock('Stores/connect.js', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => props => Component({ ...props, ...mock_props }),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+}));
+
+describe('Trade Params component', () => {
+    it('BarriersSelector component should be rendered if form_components array include it', () => {
+        const { rerender } = render(<TradeParams form_components={['barrier_selector']} is_minimized />);
+        expect(screen.getByText('Spot')).toBeInTheDocument();
+
+        rerender(<TradeParams form_components={['barrier']} is_minimized />);
+        expect(screen.queryByText('Spot')).not.toBeInTheDocument();
+    });
+});
+
+describe('Barriers Selector', () => {
+    let current_barrier;
+    beforeEach(() => {
+        render(<BarrierSelector {...mock_props} />);
+        current_barrier = screen.getByTestId('current_barrier');
+    });
+
+    it('The value of barrier_1 is chosen by default after first render', () => {
+        expect(current_barrier.innerHTML).toBe(mock_props.barrier_1);
+    });
+
+    it('The barrier list is hidden/collapsed by default', () => {
+        expect(screen.queryByText('Distance to spot')).not.toBeInTheDocument();
+    });
+
+    it('The barrier list is shown after cliking on current barrier', () => {
+        userEvent.click(current_barrier);
+
+        expect(screen.getByText('Distance to spot')).toBeInTheDocument();
+    });
+
+    it('All the available barriers from turbos_barrier_choices should be rendered in barrier list (when it can be shown)', () => {
+        userEvent.click(current_barrier);
+
+        mock_props.turbos_barrier_choices.forEach(barrier =>
+            expect(screen.getByTestId(`${barrier}`)).toBeInTheDocument()
+        );
+    });
+
+    it('After user clicked on the one of barriers option, it should be shown as current barrier value for desktop', () => {
+        userEvent.click(current_barrier);
+        const cliked_barrier = screen.getByTestId(`${mock_props.turbos_barrier_choices[1]}`);
```

Comment:
```suggestion
        const cliked_barrier = screen.getByTestId(mock_props.turbos_barrier_choices[1]);
```
@kate-deriv is mock_props.turbos_barrier_choices[1] not a string?

Reply:
Done, thank you!